core/extensions.rs

```diff
@@ -269,11 +271,12 @@ macro_rules! extension {
     $(, global_object_middleware = $global_object_middleware_fn:expr )?
     $(, external_references = [ $( $external_reference:expr ),* $(,)? ] )?
     $(, customizer = $customizer_fn:expr )?
+    $(, docs = $docblock:expr$(, $($docblocks:expr),+)?)?
     $(,)?
   ) => {
-    /// Extension struct for
-    #[doc = stringify!($name)]
-    /// .
+    $( #[doc = $docblock] )?
+    $( $( #[doc = $($docblocks)+])? )?
+    /// An extension for use with the Deno JS runtime
```

Comment:
It would not be able to be dynamic - so no names sadly
key/value attributes can't be calculated as a result of a macro at the moment in rust

We could use the workaround described [here](https://github.com/rust-lang/rust/issues/52607#issuecomment-514043687) but it seems a tad hacky to me

Reply:
We already do something like this already -- see https://docs.rs/deno_http/0.114.0/deno_http/struct.deno_http.html

Reply:
Yeah if it's own it's own line would work

It's functions like concat and stringify that don't work

Name is an ident, so we'd need stringify, no?

Reply:
Maybe it's my compiler versionm but the `#[doc = stringify!($name)` on line 275 never showed up for me

It always just said 'extension struct for .'

Has that been patched?

Reply:
Interesting! Which compiler version are you using? It definitely works on docs.rs. We don't necessarily have to use the real extension name here as long as we can make it clear in the placeholders what to write.

Reply:
cargo/rustc v1.72.0, and I've been using the latest cargo-expand to check macros

In the meantime, I've updated that block to include a better description, using extension_name as a stand-in

If it does work now, we could change `let mut extensions = vec![extension_name::init_ops_and_esm()];` in the example to using concat and stringify to sub-in the actual name of the struct?

Reply:
That would be great! If not, just using `extension_name` would be fine.

Reply:
Done! Lemme know how it looks to you,
That line still doesn't show up on my end now with the concat! usage

Reply:
Ah shit
The example doesn't compile this way
Exclude it or go back to a generic name?

Reply:
Looks like it's working well! I'll add some suggestions to tweak this a bit.

<img width="979" alt="image" src="https://github.com/denoland/deno_core/assets/512240/83f3c5fb-a8a1-442f-ba3c-2a6dc7715321">

Reply:
> Looks like it's working well! I'll add some suggestions to tweak this a bit.
> 
Ill fix the example to import runtimeoptions, on further inspection I think that's why the ci failed